midend/copyStructures.cpp

```diff
@@ -21,15 +21,19 @@ limitations under the License.
 namespace P4 {
 
 const IR::Node* RemoveAliases::postorder(IR::AssignmentStatement* statement) {
-    auto type = typeMap->getType(statement->left);
-    if (!type->is<IR::Type_StructLike>()) return statement;
+    const auto* type = typeMap->getType(statement->left);
+    if (!type->is<IR::Type_StructLike>()) {
+        return statement;
+    }
 
     ReadsWrites rw(refMap);
-    if (!rw.mayAlias(statement->left, statement->right)) return statement;
+    if (!rw.mayAlias(statement->left, statement->right)) {
+        return statement;
+    }
     auto tmp = refMap->newName("tmp");
-    auto decl = new IR::Declaration_Variable(IR::ID(tmp), type->getP4Type(), nullptr);
+    auto* decl = new IR::Declaration_Variable(IR::ID(tmp), type->getP4Type(), nullptr);
```

Comment:
I don't really think these improve the code in any way.

Reply:
This is a clang-tidy recommendation to comply with this https://llvm.org/docs/CodingStandards.html#beware-unnecessary-copies-with-auto. I do not think it can hurt to use references by default.